src/upload-lib.ts

```diff
@@ -848,3 +850,45 @@ export class InvalidSarifUploadError extends Error {
     super(message);
   }
 }
+
+function filterAlertsByDiffRange(logger: Logger, sarif: SarifFile): SarifFile {
+  const diffRanges = readDiffRangesJsonFile(logger);
+  if (!diffRanges) {
```

Comment:
An empty array is truthy in ECMAScript.  In the case that the PR has no changes, it might be better to test `diffRanges?.length` instead.

```suggestion
  if (!diffRanges?.length) {
```

Reply:
Thanks for the suggestion!  Applied.